src/Illuminate/Routing/RedirectController.php

```diff
@@ -2,20 +2,37 @@
 
 namespace Illuminate\Routing;
 
+use Illuminate\Support\Str;
+use Illuminate\Http\Request;
 use Illuminate\Http\RedirectResponse;
 
 class RedirectController extends Controller
 {
     /**
      * Invoke the controller method.
      *
-     * @param  array  $args
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Illuminate\Routing\UrlGenerator  $urlGenerator
      * @return \Illuminate\Http\RedirectResponse
      */
-    public function __invoke(...$args)
+    public function __invoke(Request $request, UrlGenerator $urlGenerator)
     {
-        [$destination, $status] = array_slice($args, -2);
+        $parameters = collect($request->route()->parameters());
 
-        return new RedirectResponse($destination, $status);
+        $status = $parameters->pop();
+
+        $destination = $parameters->pop();
+
+        $route = (new Route('GET', $destination, ['as' => 'laravel_route_redirect_destination']))->bind($request);
+
+        $parameters = $parameters->only($route->getCompiled()->getPathVariables())->toArray();
+
+        $url = $urlGenerator->toRoute($route, $parameters, false);
```

Comment:
Can we type-hint the `UrlGenerator` contract instead of this concrete implementation and call the `route` method instead of the `toRoute` method which is not on the contract?